max_iteration: 3
img_enable: 1
lidar_enable: 1

point_filter_num: 4
filter_size_surf: 0.5
filter_size_map: 0.5

common:
    lid_topic:   /points_raw
    imu_topic:   /imu_raw
    img_topic:   /kitti/camera_color_left/image_raw

preprocess:
    lidar_type: 4
    scan_line: 64
    blind: 3
    timestamp_unit: 3
    align_timestamp_en: true

mapping:
    gyr_cov: 0.05
    acc_cov: 0.1
    b_gyr_cov: 0.0001
    b_acc_cov: 0.0001
    laser_point_cov: 0.001
    det_range: 450.0
    extrinsic_est_en: false
    # extrinsic_T: [8.086759e-01, -3.195559e-01, 7.997231e-01]
    # extrinsic_R: [9.999976e-01, -7.854027e-04, 2.024406e-03,
    #               7.553071e-04,  9.998898e-01, 1.482454e-02,
    #               -2.035826e-03, -1.482298e-02, 9.998881e-01]
    extrinsic_T: [ 0, 0, 0]
    extrinsic_R: [ 1, 0, 0,
                   0, 1, 0,
                   0, 0, 1]
    Rcl: [7.53374550e-03, -9.99971465e-01, -6.16600803e-04,
          1.48024938e-02,  7.28078927e-04, -9.99890144e-01,
          9.99862007e-01,  7.52379127e-03,  1.48075522e-02]
    Pcl: [2.76988494e-02, -5.48250920e-02, -2.67312016e-01]

camera:
    debug: 0
    visual_point_cov: 1000000
    cam_width: 1242
    cam_height: 375
    cam_fx: 7.215377e+02
    cam_fy: 7.215377e+02
    cam_cx: 6.095593e+02
    cam_cy: 1.728540e+02
    grid_size: 40
    patch_size: 8

pcd_save:
    pcd_save_en: true

std:
    # pre process
    ds_size: 0.25
    maximum_corner_num: 100
    # key points
    plane_detection_thre: 0.01
    plane_merge_normal_thre: 0.2
    voxel_size: 2.0
    voxel_init_num: 10
    proj_image_resolution: 0.5
    proj_dis_min: 0
    proj_dis_max: 5
    corner_thre: 10
    # std descriptor
    descriptor_near_num: 10
    descriptor_min_len: 2
    descriptor_max_len: 50
    non_max_suppression_radius: 2
    std_side_resolution: 0.2

loop:
    pcd_save_en: true
    keyframe_dist_thresh: 1.0
    keyframe_time_thresh: 60.0
    max_loop_distance: -1
    skip_near_num: 50
    candidate_num: 10

    cdesc_verify_en: false
    cdesc_diff_thresh: 0.8
    vertex_diff_thresh: 0.5
    side_length_thresh: 0.01
    planegeo_normal_thresh: 0.2
    planegeo_dis_thresh: 0.5
    planegeo_icp_thresh: 0.4